src/Parsers/ParserCreateQuery.h

```diff
@@ -183,6 +186,11 @@ bool IParserColumnDeclaration<NameParser>::parseImpl(Pos & pos, ASTPtr & node, E
             return false;
         if (!type_parser.parse(pos, type, expected))
             return false;
+        if (s_collate.ignore(pos, expected))
+        {
+            if (!collation_parser.parse(pos, collation_expression, expected))
```

Comment:
How we should handle case when there's no keyword between `COLLATE` and `NOT NULL` ?

```
:) EXPLAIN SYNTAX CREATE TABLE t (x varchar(255) COLLATE NOT NULL) ENGINE = Memory;

EXPLAIN SYNTAX
CREATE TABLE t
(
    `x` varchar(255) NULL COLLATION NOT
)
ENGINE = Memory
```


```
:) EXPLAIN SYNTAX CREATE TABLE t (x varchar(255) COLLATE NULL) ENGINE = Memory;

EXPLAIN SYNTAX
CREATE TABLE t
(
    `x` varchar(255) COLLATION `NULL`
)
ENGINE = Memory
```

Reply:
I will check  something after the COLLATE  to make sure it is the name of validate collation (e.g.`utf8mb4_0900_ai_ci`, `latin1_german1_ci`...etc, total 273 collations got from mysql's `INFORMATION_SCHEMA.COLLATION`